### PR TITLE
[ios][web-browser] Fix bug where `safariViewControllerDidFinish` is not called

### PR DESCRIPTION
**Pull request by axeldelafosse**
# Why

To fix a bug where `safariViewControllerDidFinish` is not called if you close the webview with the "Swipe to dismiss" gesture. This bug was re-introduce by #6345 

# Test Plan

Open a web browser and try to close the webview with the "Swipe to dismiss" gesture (from left to right). You shouldn't be able to close it.

```
<View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
    <Button
      title="Open a web browser"
      onPress={() => {
        WebBrowser.openBrowserAsync('https://expo.io');
      }}
    />
</View>
```

### CODE REVIEW

**Comment by Hirbod** [comment]
I can confirm this issue. If I swipe to dismiss the browser closes instant without following the finger and can’t be reopened again without restarting the app

File: packages/expo-web-browser/ios/EXWebBrowser/EXWebBrowser.m [hunk 1]
```diff
@@ -105,13 +105,14 @@ - (dispatch_queue_t)methodQueue
...
+  // There are bugs filed already about it on OpenRadar.
+  [safariVC setModalPresentationStyle: UIModalPresentationOverFullScreen];
 
   // This is a hack to present the SafariViewController modally
   UINavigationController *safariHackVC = [[UINavigationController alloc] initWithRootViewController:safariVC];
   [safariHackVC setNavigationBarHidden:true animated:false];
-  // By setting the modal presentation style to OverFullScreen, we disable the "Swipe to dismiss"
-  // gesture that is causing a bug where sometimes `safariViewControllerDidFinish` is not called.
-  // There are bugs filed already about it on OpenRadar.
   [safariHackVC setModalPresentationStyle: UIModalPresentationOverFullScreen];
```
Comment (brentvatne) [inline packages/expo-web-browser/ios/EXWebBrowser/EXWebBrowser.m]: did you intend to move this along with the comment?

**Comment by brentvatne** [comment]
cc @mczernek

**Review by sjchmiela — APPROVED** [review]
This looks like a proper fix. The bug has been introduced in https://github.com/expo/expo/commit/3f901cf5ca6255d46b89808ceb431136e228945d, along with a fix for another bug.

**Comment by Ciberusps** [comment]
Merge? Btw copy of this issue #6918

File: packages/expo-web-browser/ios/EXWebBrowser/EXWebBrowser.m [hunk 2]
```diff
@@ -105,13 +105,14 @@ - (dispatch_queue_t)methodQueue
     safariVC.preferredControlTintColor = [EXWebBrowser convertHexColorString:arguments[WebBrowserControlsColorKey]];
   }
   safariVC.delegate = self;
+  // By setting the modal presentation style to OverFullScreen, we disable the "Swipe to dismiss"
+  // gesture that is causing a bug where sometimes `safariViewControllerDidFinish` is not called.
+  // There are bugs filed already about it on OpenRadar.
+  [safariVC setModalPresentationStyle: UIModalPresentationOverFullScreen];
```
Comment (sjchmiela) [inline packages/expo-web-browser/ios/EXWebBrowser/EXWebBrowser.m]: This reverts https://github.com/expo/expo/commit/3f901cf5ca6255d46b89808ceb431136e228945d, which fixed fullscreen browser being presented in an iOS 13 modal. Could you please verify that this no longer happens? Or maybe we need to set this `modalPresentationStyle` on both `safariVC` and `safariHackVC`?

Comment (axeldelafosse) [reply packages/expo-web-browser/ios/EXWebBrowser/EXWebBrowser.m]: Yes. The browser is presented fullscreen. We need to set this style on both VC

File: packages/expo-web-browser/ios/EXWebBrowser/EXWebBrowser.m [hunk 3]
```diff
@@ -105,13 +105,14 @@ - (dispatch_queue_t)methodQueue
...
 
   // This is a hack to present the SafariViewController modally
```
Comment (brentvatne) [inline packages/expo-web-browser/ios/EXWebBrowser/EXWebBrowser.m]: not related to this particular PR but we should include more context in comments like this, and err on the side of providing too much information. i don't personally understand the history behind this, if anyone does know please augment the comment :)

File: packages/expo-web-browser/ios/EXWebBrowser/EXWebBrowser.m (hunk 2 above)
Comment (sjchmiela) [reply packages/expo-web-browser/ios/EXWebBrowser/EXWebBrowser.m]: Ah, ok, sorry, now I see that the style is set on both VCs. 👍

Comment (axeldelafosse) [reply packages/expo-web-browser/ios/EXWebBrowser/EXWebBrowser.m]: No problem 😊Thank you

**Pull request merged**